CONTRIBUTING.md

```diff
@@ -325,6 +326,84 @@ if you change the history (rebase, squash, amend), and use git push --force to
 update the branch on your fork. The pull request points to that branch, not to
 specific commits.
 
+## Exercism.io Web Structure
+
+### Intro
+Exercism is built with the [Sinatra](https://github.com/sinatra/sinatra) web framework.
+If you haven't heard of it, definitely check it out. It's an awesome lightweight web frameworks for Ruby.
+
+Exercism seems to follow what some may call an MVP (Model-View-Presenter) architecture.
+Read more about that [here](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93presenter).
+
+### Base Layout (Where the Magic Happens)
+
+#### App (Routes, Presenters and Views)
+App is the user facing side of exercism.
+It handles logging in (and other functions), making calls out to the database to store or get information,
+structuring HTML (that can later be styled), etc.
+
+**Helpers** can be anything that *helps* presentation of data. For exercism, these are things like:
+wrappers for Markdown and Syntax Highlighting parsers, session handling, etc.
+Helpers are located in `app/helpers.rb` If you add a helper, update the list of existing helpers in this file (the convention is `ClassName: 'filename'`).
+
+**Presenters** are for showing information that would be beneficial to users or the views but maybe not the best way to store data.
+For example, in exercism, most times are stored in UTC (generally most servers store time this way) but... users might want to see times in their own timezone.
+Presenters can take relevant information (like showing comment notifications on the dashboard) and transform it to be more personal.
+Presenters are located in `app/presenters.rb`. If you add a presenter, update the list of existing presenters in this file (the convention is `ClassName: 'filename'`).
+
+**Routes** in Sinatra are kind of like a combination of routes and controllers in Rails.
+In a route, you can specify endpoints (like `exercism.io/login`) and define how the app should respond (i.e. accept some credentials and log the user in or handle errors accordingly).
+Routes are located in `app/routes.rb`. If you add a route, update the list of existing routes in this file (the convention is `ClassName: 'filename'`).
+Routes are unique in that you'll also have to update the main `app.rb` with the new route info (the convention is `Routes::ClassName`).
```

Comment:
We might want to say something about how each of the app/routes/*.rb files contain separate sinatra apps that all inherit from Core (app/routes/core.rb). Then app.rb contains the top-level app that defines all of the other sub-apps as rack middleware.